### ISSUE TITLE
About accuracy and bug.

### ISSUE THREAD

**Issue by rui-1997** [dms-net/scatteringGCN#2]
(1) I think there is a bug in your code. The scattering matrix of order 3 is wrongly calculated. The code you released computes (P^3-I)P^3, but it should be P^4. 

(2) Regarding the accuracy, I have run your codes and find that ScatteringGCN performs poorly no matter in random splits or the standard splits. It is even outperformed by the vanilla GCN in almost all cases.

**Comment by yimengmin** [comment]
Hi  yangrui1997 
Thanks.
'scattering matrix of order 3', yes, the psi_3=scattering1st(adj,3) scattering matrix gives you P^3, for P^4 psi_3 should be scattering1st(adj,4). We change that to scattering1st(adj,4) now.
https://github.com/dms-net/scatteringGCN/blob/5445ffbd4dff1c4bf089040006e827481fc5b793/utils.py#L135

Regarding the accuracy, can you please try to play with the hyperparameters? e.g. smoo/hid1/hid2? I just ran the code: 'python train.py' on NVIDIA V100 and the accuracy gives me 83.6% on the default cora dataset. And if you go to the DBLP dir run DBLP_SCT.py and DBLP_GCN.py, the accuracy gap is very large.

**Comment by yangrui1997** [comment]
Could you give the real hyperparameter setting on Cora, Citeseer, and Pubmed? I ran the code with the hyperparameter setting reported in the appendix of your paper, but cannot get good performance. Moreover, with random dataset splits suggested by "Pitfalls of Graph Neural Networks Evaluation", the performance is even worse. Moreover, I have tried Scattering GCN on Amazon Photo and Amazon Computers, the accuracy was worse than the baselines including the vanilla GCN in a large margin.
So, I really hope you can provide the real hyperparameter setting.
Thanks!

**Comment by yimengmin** [comment]
the default parameters are in the train.py so you can just run 
'''
python train.py
'''
btw I am not surprised that the accuracy is different following the 'using the Pitfalls of Graph Neural Network Evaluation'. 
The structure is very sensitive to the width of different channels. So if you use a different spilt you HAVE TO grid search the parameters again. We don't use the parameters in the appendix because the GitHub version seems more robust on different environments.

Back to the paper 'using the Pitfalls of Graph Neural Network Evaluation'. I think this problem raised by the paper(sensitive to the dataset spilt/hyperparameter)  can be attributed to some intrinsic problems of Cora, CiteSeer and PubMed. These three datasets should not been used as a baseline for judging GNN architectures. You can get any results on these three datasets. For further demonstrating our pipeline is better than GCN/GAT we use Wiki-CS dataset with 20 different splints.(https://arxiv.org/abs/2007.02901)

**Comment by yangrui1997** [comment]
Overall, I do not agree with your comments.

Firstly, I think you commet that "These three datasets (Cora, Citeseer and Pubmed) should not been used as a baseline for judging GNN architectures." is not academically rigorous. Do you mean that all the papers that use these three datasets are not convincing? Moreover, if your method does not perform well on these datasets, why you put the results on your papers and give wrong hyperparameter setting in the appendix? 

Secondly, your codes only give the default setting for Cora, is the other datasets the same? And what is the search space to do grid search? I just ran your codes on Amazon Photo and Amazon Computers, Scattering GCN performs poorly (67.00% (scattering gcn) vs 82.35% (vanilla gcn) on Amazon Computers).

Thirdly, I do not see any results of Scattering GCN on Wiki-CS with 20 different splits and your code cannot reproduce corresponding results. Moreover, outperforming GCN and GAT does not mean it can achieve SOTA, many methods have outperformed these two methods by a large margin.

Thanks.

**Comment by yangrui1997** [comment]
Your codes have not fixed the bug. Regarding this issue and your statement that the setting in your appendix is wrong, I am skeptical about the results reported in your paper.
Thanks.

**Comment by mrguywolf** [comment]
This is getting a bit off topic.

For your initial report - thank you for raising the issue with P^3 vs. P^4 - looking at the code, this is now fixed. Regarding accuracy - as Yimeng said (and as we say in the paper), the performance depends on hyperparameter tuning, as is the case for most if not all deep learning methods, which is a pretty standard practice in ML. We use a standard grid search (described in details in the supplement of our paper), but one can also use other values - in our experience, the general trend shown in our results is not difficult to reproduce and we have already seen others also recreate these. Without seeing the specific configuration and hyperparameters you are using it is difficult to address this further.

Regarding other topics:
1. SOTA - we never claim anywhere in the paper that we establish SOTA performance, but rather that we establish the benefits of adding scattering channels in a hybrid architecture together with GCN channels, which alleviates oversmoothing. These channels can also be added in various other architectures, and whether or not they can achieve SOTA on appropriate datasets (see below - Cora, Citeseer, and Pubmed are hardly sufficient for or reflective of SOTA capabilities) is left for future work looking into such more intricate architectures.
2. Datasets - the three "classic" citation datasets (Cora, Citeseer, Pubmed) are extremely simple and very far from reflecting real challenges in graph representation learning or node classification. Indeed, they mostly provide a sanity check at the level of a toy example. This is well accepted in the GRL community, so to directly answer your question - yes, from an academically rigorous point of view, any method that relies solely on these datasets to claim SOTA is dubious at best. From my experience, most reviewers in the field would be rather underwhelmed if not completely unconvinced by comparisons on these datasets alone, without considering other data, and have been for quite some time now (at least a year... maybe more).
3. Other splits - we followed the standard practice in the splits we used in order to provide comparable results to those shown in recent literature. It should be noted that there are multiple different split configurations on these three datasets providing a plethora of inconsistent and/or conflicting results - for example, there are quite a few works where GAT shows up in tables as being significantly outperformed by GCN, while other work show the opposite. Dealing with such discrepancies within the GRL community is out of scope for this work, and in any case unrelated to the hybrid architecture proposed here to augment GCN.
4. Other datasets - since we have not tested the method on these Amazon datasets I cannot comment on them, but I would note that all the information extracted in GCN is still available to the hybrid model - we are just adding information with the scattering channels - so one would expect that in the worst case (if the additional channels are unhelpful), this would just fall back to work like vanilla GCN. Regardless, we do see significant improvements in various other datasets, including DBLP shown in the NeurIPS paper and some others (incl. Wiki-CS) that appear in follow up work on geometric scattering attention networks.
5. While the code here does not include Wiki-CS, it is not difficult to apply it to various datasets and perform a grid search to properly tune hyperparameters. More advanced versions with other data will likely appear in the future either on this repository or others.

Thank you, and you are more than welcome to contact us by email to continue this discussion and ask for further guidance and help with using our code.
